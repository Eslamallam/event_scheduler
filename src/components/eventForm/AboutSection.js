import React from "react";
import { categories } from "../../data/categories";

export const AboutSection = ({
  handleChange,
  inputValues,
  errors,
  chars,
  maxChars,
}) => {
  return (
    <div className="form-section">
      <h2 className="separator form-heading">About</h2>

      <div className="form-control">
        <label>title: *</label>
        <input
          className={`control ${errors.title ? "input-error" : null}`}
          type="text"
          name="title"
          placeholder="Make it short and clear"
          onChange={handleChange}
          value={inputValues.title}
        />
        {errors.title && <span className="error">{errors.title}</span>}
      </div>

      <div className="form-control">
        <label>description: *</label>
        <div className="text-area-box">
          <textarea
            type="text"
            name="description"
            className={`control-area ${
              errors.description ? "input-error" : null
            }`}
            placeholder="write about your event, be creative"
            onChange={handleChange}
            value={inputValues.description}
            rows="10"
            maxLength={maxChars}
          ></textarea>
          <div className="text-counter">
            <p>Max characters allowed</p>
          </div>
          <div className="counter">
            <p>
              {chars.charsLeft}
              {`/${maxChars}`}
            </p>
          </div>
        </div>

        {errors.description && (
          <span className="error">{errors.description}</span>
        )}
      </div>

      <div className="form-control">
        <label>category:</label>
        <select onChange={handleChange} name="category_id" className="control">
          <option value={""}>Select Category</option>
          {categories.map((category) => {
            return (
              <option key={category.id} value={category.id}>
                {category.name}
              </option>
            );
          })}
        </select>
      </div>

      <div className="form-control">
        <label>payment:</label>
        <div className="radio-group">
          <input
            className="radio-control"
            type="radio"
            name="event_type"
            onChange={handleChange}
            value="free"
          />
          Free event
          <input
            className="radio-control"
            type="radio"
            name="event_type"
            onChange={handleChange}
            value="paid"
          />
          Paid event
        </div>

        {inputValues && inputValues.event_type === "paid" ? (
          <div className="fee">
            <input
              className={`sm-control ${
                errors.event_fee ? "input-error" : null
              }`}
              type="number"
              name="event_fee"
              min={0}
              placeholder="Number"
              onChange={handleChange}
              value={inputValues.event_fee}
            />
            $
            {errors.event_fee && (
              <span className="error">{errors.event_fee}</span>
            )}
          </div>
        ) : null}
      </div>

      <div className="form-control">
        <label>reward:</label>
        <input
          className="sm-control"
          type="number"
          name="reward"
          min={0}
          placeholder="Number"
          onChange={handleChange}
          value={inputValues.reward}
        />
        <p className="text">reward points for attendance</p>
      </div>
    </div>
  );
};
