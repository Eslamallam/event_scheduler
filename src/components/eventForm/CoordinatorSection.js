import React from "react";
import { coordinators } from "../../data/coordinators";

export const CoordinatorSection = ({ handleChange, inputValues, errors }) => {
  return (
    <div className="form-section">
      <h2 className="separator form-heading">Coordinator</h2>

      <div className="form-control">
        <label>responsible: *</label>
        <select
          onChange={handleChange}
          name="coordinator"
          className={`control ${errors.coordinator ? "input-error" : null} `}
        >
          <option value={""}>Select Responsible</option>
          {coordinators.map((coordinator) => {
            return (
              <option
                key={coordinator.id}
                value={`${coordinator.email}-${coordinator.id}`}
              >
                {coordinator.name} {coordinator.lastname}
              </option>
            );
          })}
        </select>

        {errors.coordinator && (
          <span className="error">{errors.coordinator}</span>
        )}
      </div>

      <div className="form-control">
        <label>email: </label>
        <input
          className={`control ${errors.email ? "input-error" : null}`}
          type="email"
          name="email"
          placeholder="Email"
          onChange={handleChange}
          value={inputValues.email}
        />
        {errors.email && <span className="error">{errors.email}</span>}
      </div>
    </div>
  );
};
