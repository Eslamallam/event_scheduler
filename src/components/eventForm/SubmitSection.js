import React from "react";

export const SubmitSection = () => {
  return (
    <div className="submit">
      <button type="submit">publish event</button>
    </div>
  );
};
