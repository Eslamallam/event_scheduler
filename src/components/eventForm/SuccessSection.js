import React from "react";

export const SuccessSection = () => {
  return (
    <div className="success-container">
      <h2>Success</h2>
      <p>Event has been created</p>
    </div>
  );
};
