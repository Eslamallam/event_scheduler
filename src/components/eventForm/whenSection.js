import React from "react";

export const WhenSection = ({ handleChange, inputValues, errors }) => {
  return (
    <div className="form-section">
      <h2 className="separator form-heading">When</h2>

      <div className="form-control">
        <label>starts on: *</label>
        <input
          className={`date-control ${errors.date ? "input-error" : null} `}
          type="date"
          name="date"
          onChange={handleChange}
        />
        <span className="text-separator">at</span>
        <input
          className={`time-control ${errors.date ? "input-error" : null} `}
          type="time"
          name="event_time"
          onChange={handleChange}
        />

        {errors.date && <span className="error">{errors.date}</span>}
      </div>

      <div className="form-control">
        <label>duration:</label>
        <input
          className="sm-control"
          type="number"
          name="duration"
          min={1}
          placeholder="Number"
          onChange={handleChange}
          value={inputValues.duration}
        />
        <p className="text">hour</p>
      </div>
    </div>
  );
};
