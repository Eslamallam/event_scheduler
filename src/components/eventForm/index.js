import React from "react";
import { useForm } from "../../helpers/useForm";
import { validate } from "../../helpers/validate";
import * as actions from "../../store/actions";
import { useDispatch } from "react-redux";
import { AboutSection } from "./AboutSection";
import { CoordinatorSection } from "./CoordinatorSection";
import { WhenSection } from "./whenSection";
import { SubmitSection } from "./SubmitSection";
import { SuccessSection } from "./SuccessSection";

export const EventForm = () => {
  const {
    handleChange,
    handleSubmit,
    values,
    errors,
    chars,
    maxChars,
    showSuccess,
  } = useForm(formSubmit, validate);

  const dispatch = useDispatch();

  function formSubmit() {
    const formData = {};

    // data to be submitted
    formData.title = values.title;
    formData.description = values.description;
    formData.category_id = parseInt(values.category_id) || null;
    formData.paid_event = values.event_type === "paid" ? true : false;
    formData.event_fee = parseInt(values.event_fee) || null;
    formData.reward = parseInt(values.reward) || null;
    formData.date = values.date + "T" + values.event_time;
    formData.duration = parseInt(values.duration) * 60 * 60 || null;
    formData.coordinator = {
      email: values.coordinator.split("-")[0],
      id: values.coordinator.split("-")[1],
    };

    console.log(formData);
    dispatch(actions.doCreateEvent(formData));
  }

  return (
    <div className="event">
      {showSuccess ? (
        <SuccessSection />
      ) : (
        <form onSubmit={handleSubmit} noValidate>
          <AboutSection
            handleChange={handleChange}
            inputValues={values}
            errors={errors}
            chars={chars}
            maxChars={maxChars}
          />
          <CoordinatorSection
            handleChange={handleChange}
            inputValues={values}
            errors={errors}
          />
          <WhenSection
            handleChange={handleChange}
            inputValues={values}
            errors={errors}
          />

          <SubmitSection />
        </form>
      )}
    </div>
  );
};
