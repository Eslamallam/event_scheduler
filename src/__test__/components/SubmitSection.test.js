import React from "react";
import Adapter from "enzyme-adapter-react-16";
import { shallow, configure } from "enzyme";
import toJSON from "enzyme-to-json";
import { SubmitSection } from "../../components/eventForm/SubmitSection";

configure({ adapter: new Adapter() });

test("should render Success section correctly", () => {
  const wrapper = shallow(<SubmitSection />);
  expect(wrapper.find("button").text()).toBe("publish event");
  expect(toJSON(wrapper)).toMatchSnapshot();
});
