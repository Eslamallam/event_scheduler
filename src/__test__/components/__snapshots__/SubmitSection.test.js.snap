// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Success section correctly 1`] = `
<div
  className="submit"
>
  <button
    type="submit"
  >
    publish event
  </button>
</div>
`;
